src/nvim/terminal.c

```diff
@@ -442,7 +442,7 @@ void terminal_enter(void)
   RedrawingDisabled = s->save_rd;
   apply_autocmds(EVENT_TERMLEAVE, NULL, NULL, false, curbuf);
 
-  if (save_curwin == curwin) {  // save_curwin may be invalid (window closed)!
+  if (save_curwin == curwin->handle) {
```

Comment:
the comment is still relevant: the reason we are checking this is because we don't want to enter this block if the window was closed.
```suggestion
  if (save_curwin == curwin->handle) {  // Else: window was closed.
```